thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -3103,7 +3117,8 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                     Owned<IFileIO> iFileIO = createIFileI(lenArray.item(p), tlkMb.toByteArray()+posArray.item(p));
                     StringBuffer name("TLK");
                     name.append('_').append(container.queryId()).append('_');
-                    Owned<IKeyIndex> tlkKeyIndex = createKeyIndex(name.append(p).str(), 0, *iFileIO, (unsigned) -1, true, 0); // MORE - not the right crc
+                    size32_t bufferSize = getBlockedRandomBufferSize(iFileIO);
```

Comment:
this change doesn't make sense.

The iFileIO it's reading here (declared a few lines above), is a IFileIO constructed from memory, not disk, so it does not make sense to pass through the blocked random io size - which is a plane thing - for reading from disk.
This change should be reverted.